app/controllers/patients_controller.rb

```diff
@@ -542,6 +548,105 @@ def update_monitoring_fields(patient, params, household, propagation)
     patient.monitoring_history_edit(history_data, diff_state)
   end
 
+  # Update the patient's follow-up flag fields
+  def update_follow_up_flag
+    redirect_to(root_url) && return unless current_user.can_edit_patient? && !current_user.role?(Roles::ENROLLER)
```

Comment:
I think the following 2 checks are already done in the `bulk_update` method so it seems kind of redundant here:
```
redirect_to(root_url) && return unless current_user.can_edit_patient?
```
as for checking for the user to not be an enroller, I think we can just make a new user method in `user.rb` called `can_bulk_edit_patients?`, define the roles that are allowed to bulk edit, and then in `bulk_update`, we can do:
```
redirect_to(root_url) && return unless current_user.can_bulk_edit_patients?
```

Reply:
good question, I believe the `allowed_params` there are only used for creating and editing patients via the enrollment wizard, so those params would only include fields relevant to enrollment